services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/HankeServiceITests.kt

```diff
@@ -894,61 +894,82 @@ class HankeServiceITests : DatabaseTest() {
-                    applicationData =
-                        AlluDataFactory.createCableReportApplicationData(name = tooLongName)
-                )
+        @Test
+        fun `generates hanke based on application`() {
+            val inputApplication = AlluDataFactory.cableReportWithoutHanke()
+
+            val application = hankeService.generateHankeWithApplication(inputApplication, USER_NAME)
+
+            assertThat(application.applicationData.name)
+                .isEqualTo(inputApplication.applicationData.name)
+            val hanke = hankeRepository.findByHankeTunnus(application.hankeTunnus)!!
+            assertThat(hanke.generated).isTrue
+            assertThat(hanke.status).isEqualTo(HankeStatus.DRAFT)
+            assertThat(hanke.hankeTunnus).isEqualTo(application.hankeTunnus)
+            assertThat(hanke.nimi).isEqualTo(application.applicationData.name)
+            assertThat(hanke.perustaja?.nimi).isEqualTo("Teppo Testihenkilö")
+            assertThat(hanke.perustaja?.email).isEqualTo("teppo@example.test")
+        }
 
-        val application = hankeService.generateHankeWithApplication(inputApplication, USER_NAME)
+        @Test
+        fun `sets hanke name according to limit and saves successfully`() {
+            val expectedName = "a".repeat(MAXIMUM_HANKE_NIMI_LENGTH)
+            val tooLongName = expectedName + "bbb"
+            val inputApplication =
+                AlluDataFactory.cableReportWithoutHanke()
+                    .copy(
+                        applicationData =
+                            AlluDataFactory.createCableReportApplicationData(name = tooLongName)
+                    )
+
+            val application = hankeService.generateHankeWithApplication(inputApplication, USER_NAME)
+
+            val hanke = hankeRepository.findByHankeTunnus(application.hankeTunnus)!!
+            assertThat(hanke.nimi).isEqualTo(expectedName)
+            assertThat(hanke.generated).isTrue
+            assertThat(hanke.status).isEqualTo(HankeStatus.DRAFT)
+            assertThat(hanke.hankeTunnus).isEqualTo(application.hankeTunnus)
+            assertThat(hanke.perustaja?.nimi).isEqualTo("Teppo Testihenkilö")
+            assertThat(hanke.perustaja?.email).isEqualTo("teppo@example.test")
+        }
 
-        val hanke = hankeRepository.findByHankeTunnus(application.hankeTunnus)!!
-        assertThat(hanke.nimi).isEqualTo(expectedName)
-        assertThat(hanke.generated).isTrue
-        assertThat(hanke.status).isEqualTo(HankeStatus.DRAFT)
-        assertThat(hanke.hankeTunnus).isEqualTo(application.hankeTunnus)
-        assertThat(hanke.perustaja?.nimi).isEqualTo("Teppo Testihenkilö")
-        assertThat(hanke.perustaja?.email).isEqualTo("teppo@example.test")
-    }
+        @Test
+        fun `creates hankealueet from the application areas`() {
+            val inputApplication = AlluDataFactory.cableReportWithoutHanke()
+            assertThat(inputApplication.applicationData.areas).isNotNull.isNotEmpty
```

Comment:
Would this work without the null check? I mean isNotEmpty should should fail for nulls.

Same thing for features size check.

Reply:
I was worried they wouldn't give a good error message in the null case. But they have an explicit null assert buried inside, so my worries were unfounded. Looks like I'm already starting to forget how AssertJ works. I dropped the null asserts.